src/arithmetic.jl

```diff
@@ -0,0 +1,56 @@
+Base.promote_rule(::Type{Arf}, ::Type{<:Union{AbstractFloat,Integer}}) = Arf
+Base.promote_rule(::Type{Arb}, ::Type{<:Union{AbstractFloat,Integer,Arf,ArbRef}}) = Arb
+Base.promote_rule(
+    ::Type{Acb},
+    ::Type{<:Union{AbstractFloat,Integer,Complex,Arf,Arb,ArbRef,AcbRef}},
+) = Acb
+
+for (jf, af) in [(:+, :add!), (:-, :sub!), (:*, :mul!), (:/, :div!)]
+    @eval function Base.$jf(x::T, y::T) where {T<:Union{Arf,Arb,ArbRef,Acb,AcbRef}}
+        z = T(prec = max(precision(x), precision(y)))
+        $af(z, x, y)
+        z
+    end
+end
+function Base.:(-)(x::T) where {T<:Union{Arf,Arb,ArbRef,Acb,AcbRef}}
+    z = T(prec = precision(x))
+    neg!(z, x)
+    z
+end
+function Base.:(^)(x::T, k::Integer) where {T<:Union{Arf,Arb,ArbRef,Acb,AcbRef}}
+    z = T(prec = precision(x))
+    pow!(z, x, convert(UInt, k))
+    z
+end
+
+for f in [
+    :sqrt,
+    :exp,
+    :log,
+    :log1p,
+    :sin,
+    :cos,
+    :tan,
+    :cot,
+    :asin,
+    :acos,
+    :atan,
+    :acot,
+    :sinc,
+    :sinh,
+    :cosh,
+    :tanh,
+    :asinh,
+    :acosh,
+    :atanh,
+    :sec,
+    :asec,
+    :sech,
+    :asech,
+]
+    @eval function Base.$f(x::T) where {T<:Union{Arf,Arb,ArbRef,Acb,AcbRef}}
+        z = T(prec = max(precision(x)))
+        $(Symbol(f, :!))(z, x)
+        z
+    end
+end
```

Comment:
`Arf` implements surprisingly few functions, basically only addition, subtraction, multiplication, division and square root. In particular it doesn't implement any trigonometric functions. Since they are so limited we should maybe consider treating `Arf` more of an internal type, similar to `Mag`. We could of course convert to `Arb`, preform the operation, and convert back (this is how [ArbFloats.jl](https://github.com/JuliaArbTypes/ArbFloats.jl) does it) but that sort of destroys the point.

Reply:
`Arf` is really an add one. I will just remove `Arf` from the list, and I think the conversion back and forth should be done explicitly.